src/miner.cpp

```diff
@@ -183,6 +186,9 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const CScript& s
 
             const CTransaction& tx = iter->GetTx();
 
+            if (!fIncludeWitness && !tx.wit.IsNull())
```

Comment:
This isn't necessary right?  Perhaps we need a better way to identify which rules for block assembly are enforced by the mempool, but I'd rather that we keep redundant checks out to make future work on block assembly easier.  At the very least this should be commented that unless there is a reorg to a lower height more work chain immediately at the segwit activation border, then this check is superfluous.